spec/integ/matrix-client-event-timeline.spec.ts

```diff
@@ -1624,10 +1626,35 @@ describe("MatrixClient event timelines", function () {
                     },
                 },
             });
+            httpBackend
```

Comment:
Please could you explain to me how these changes to the test check we are now behaving correctly?

I was expecting to see a test which failed before because events were out of order, but now passes.

Reply:
The issue with that is that the tests often omit certain parts of the way threads are loaded, to reduce complexity. Otherwise our tests would be much larger than the are. Ideally, each of these tests only tests the part of the functionality that it's concerned with, and mocks the rest.

But as this PR changes a minor part of the original functionality, the way tests mock the loading has to be adapted.

There's no single test that can easily test this issue yet, because the way we mock the loading can mask the issue itself.

Reply:
We talked over Matrix and agreed that @justjanne will work on a test that failed before and passes now.

The new test will either be part of this PR, or a new one as soon as possible afterwards.